torchtext/experimental/datasets/sequence_tagging.py

```diff
@@ -0,0 +1,220 @@
+    else:
+        if not isinstance(vocabs, list):
+            raise TypeError("vocabs must be an instance of list")
+
+        # Find data that's not None
+        notnone_data = None
+        for key in raw_data.keys():
+            if raw_data[key] is not None:
+                notnone_data = raw_data[key]
+                break
+        if len(vocabs) != len(notnone_data[0]):
+            raise ValueError(
+                "Number of vocabs must match the number of columns "
+                "in the data")
+
+    if word_transform:
+        word_transform = sequential_transforms(word_transform,
+                                               vocab_func(vocabs[0]),
+                                               totensor(dtype=torch.long))
+    else:
+        word_transform = sequential_transforms(vocab_func(vocabs[0]),
+                                               totensor(dtype=torch.long))
+    labels_transforms = [
+        sequential_transforms(vocab_func(vocabs[idx + 1]),
+                              totensor(dtype=torch.long))
+        for idx in range(len(vocabs) - 1)
+    ]
+    transformers = [word_transform, *labels_transforms]
+
+    datasets = []
+    for item in data_select:
+        if raw_data[item] is not None:
+            datasets.append(
+                SequenceTaggingDataset(raw_data[item], vocabs, transformers))
+
+    return datasets
+
+
+class SequenceTaggingDataset(torch.utils.data.Dataset):
+    """Defines an abstraction for raw text sequence tagging iterable datasets.
+    Currently, we only support the following datasets:
+        - UDPOS
+        - CoNLL2000Chunking
+    """
+    def __init__(self, data, vocab, transforms):
+        """Initiate sequence tagging dataset.
+        Arguments:
+            data: a list of word and its respective tags. Example:
+                [[word, POS, dep_parsing label, ...]]
+            vocab: Vocabulary object used for dataset.
+            transforms: a list of string transforms for words and tags.
+        """
+
+        super(SequenceTaggingDataset, self).__init__()
+        self.data = data
+        self.vocab = vocab
+        self.transforms = transforms
+
+    def __getitem__(self, i):
+        line = []
+        for idx, transform in enumerate(self.transforms):
+            line.append(transform(self.data[i][idx]))
+        return line
+
+    def __len__(self):
+        return len(self.data)
+
+    def get_vocab(self):
+        return self.vocab
+
+
+def UDPOS(train_filename="en-ud-tag.v2.train.txt",
+          valid_filename="en-ud-tag.v2.dev.txt",
+          test_filename="en-ud-tag.v2.test.txt",
+          data_select=("train", "valid", "test"),
+          root=".data",
+          vocabs=None,
+          word_tokenizer=None):
+    """ Universal Dependencies English Web Treebank
+
+    Separately returns the training and test dataset
+
+    Arguments:
+        train_filename: Filename for training dataset.
```

Comment:
Same like the above, do we need those filenames?

Reply:
removed